apps/platform/trpc/routers/authRouter/totpRouter.ts

```diff
@@ -0,0 +1,92 @@
+import { z } from 'zod';
+import { router, userProcedure } from '../../trpc';
+import { eq } from '@uninbox/database/orm';
+import { accounts, users } from '@uninbox/database/schema';
+import { decodeHex, encodeHex } from 'oslo/encoding';
+import { TOTPController, createTOTPKeyURI } from 'oslo/otp';
+import { TRPCError } from '@trpc/server';
+
+export const totpRouter = router({
+  createTotpSecret: userProcedure
+    .input(z.object({}).strict())
+    .mutation(async ({ ctx }) => {
+      const { user, db } = ctx;
+      const userId = user.id;
+      const existingData = await db.query.users.findFirst({
+        where: eq(users.id, userId),
+        columns: {
+          username: true
+        },
+        with: {
+          account: {
+            columns: {
+              totpSecret: true
+            }
+          }
+        }
+      });
+      if (existingData.account.totpSecret) {
+        throw new TRPCError({
+          code: 'BAD_REQUEST',
+          message: 'TOTP secret already exists'
+        });
+      }
+      const newSecret = crypto.getRandomValues(new Uint8Array(20));
```

Comment:
can we use anything other than crypto?
maybe nanoId?

Reply:
olso suggests it, nanoId uses crypto under the hood anyways

Reply:
the secret is a `Uint8Array`, if we use nanoId then we will also need to convert it back and forth, so I think its fine to use crypto while dealing with non string stuff